services/transactionService.ts

```diff
@@ -163,45 +113,26 @@ export async function upsertManyTransactionsForAddress (transactions: Transactio
   return ret.filter((t) => t !== undefined) as TransactionWithAddressAndPrices[]
 }
 
-export async function syncTransactionsForAddress (addressString: string): Promise<TransactionWithAddressAndPrices[]> {
+export async function syncTransactionsForAddress (addressString: string, start: number = 0, maxTransactions: number = Infinity): Promise<TransactionWithAddressAndPrices[]> {
   const address = await fetchAddressBySubstring(addressString)
-  let newTransactionsCount = -1
-  let seenTransactionsCount = 0
-  let insertedTransactions: TransactionWithAddressAndPrices[] = []
-  while (newTransactionsCount !== 0) {
-    const addressTransactions = await getAddress({
-      address: address.address,
-      nbFetch: FETCH_N,
-      nbSkip: seenTransactionsCount
-    })
-    let nextConfirmedTransactions = addressTransactions.confirmedTransactionsList
-    let nextUnconfirmedTransactions = addressTransactions.unconfirmedTransactionsList.map(mempoolTx => parseMempoolTx(mempoolTx))
-
-    // remove transactions older than the networks
-    nextConfirmedTransactions = nextConfirmedTransactions.filter(txThesholdFilter(address))
-    nextUnconfirmedTransactions = nextUnconfirmedTransactions.filter(txThesholdFilter(address))
 
-    newTransactionsCount = nextConfirmedTransactions.length // Unconfirmed transactions are NOT taken into account when skipping with nbSkip
-    seenTransactionsCount += newTransactionsCount
+  const addressTransactions = await getAddressTransactions(addressString, start, maxTransactions)
+  const insertedTransactions: TransactionWithAddressAndPrices[] = [
+    ...await upsertManyTransactionsForAddress(addressTransactions.confirmed, address),
+    ...await upsertManyTransactionsForAddress(addressTransactions.unconfirmed, address)
+  ]
 
-    let newInsertedTransactions = await upsertManyTransactionsForAddress(nextConfirmedTransactions, address)
-    newInsertedTransactions = [
-      ...newInsertedTransactions,
-      ...(await upsertManyTransactionsForAddress(nextUnconfirmedTransactions, address, false))
-    ]
-    insertedTransactions = [...insertedTransactions, ...newInsertedTransactions]
-    await new Promise(resolve => setTimeout(resolve, FETCH_DELAY))
-  }
+  if (maxTransactions === Infinity || addressTransactions.confirmed.length < maxTransactions) { await updateLastSynced(addressString) } else { void syncTransactionsForAddress(addressString, start + addressTransactions.confirmed.length) }
```

Comment:
1. This could be broken into more lines for better redability
2. The recursive call to `syncTransactionsForAddress` will always use `maxTransactions === Infinity`. I actually think we should avoid anywhere for the cost of memory and simply because of having to wait for ALL transactions to be fetched until we see the next one in the DB
3. The function should receive a `parameters: GetAddressTransactionsResponse`

Reply:
1. indeed, linter did that, not me, probably because I wrote the `if` without `{}`, now I will leave them there to see if it doesn't change to a one liner again
2. refactored so that each block of 100 is read from the blockchain and persisted in the database each interaction and not kept in memory for the next 100 read, like it was before this PR (but instead of all this happening in TransactionService, now it happens inside grpc, TransactionService has a much cleaner logic now and grpc types stay encapsulated inside grpc file, which is totally worth the price of a bit of logic duplication between chronik and grpc which can be addressed later when chronik is implemented)
3. done (`GetAddressTransactionsParameters` actually)